### ISSUE TITLE
Missing Data and the question of implementing a hypergeometric projection method

### ISSUE THREAD

**Issue by isaacovercast** [popgenmethods/momi2#7]
Ok, here's a real question. Lately I've been guiding people away from a certain simulation based demographic inference program that shall remain nameless, and have been recommending momi2. The simplicity of the API interface, the nice tutorial docs, and the ease of plotting the demographic model are killer features. Being able to construct a model and then visually evaluate whether it matches your expectation is really useful. Now here's the thing, most of the people I work with are generating RAD-Seq type datasets which are characterized by moderate to high levels of missing data. As far as I understand, when momi2 constructs the SFS it only includes sites that are fully represented in the sample (which obviously makes sense because of the nature of the SFS). I've been _meaning_ to port over the hypergeometric projection method implemented in dadi, but I'm wondering whether you haven't thought of this already and haven't implemented it for a good reason.

Alternatively, people could perform the projection in dadi and then import the SFS into momi2, but I can't find a function that seems appropriate for this. Maybe I just haven't looked hard enough, but it seems that there isn't a way to load an SFS from a dadi style SFS vector or joint SFS matrix. Am I missing something?

Long story short, I'd love to get non-model folks using momi2 more, I'm encouraging them to do so, but I'm kind of withholding this last little bit of info that the momi2 SFS is (often) drastically downsampling their data because of the sparsity of their data matrices (again, unless I'm missing something obvious). Got any specific guidance here? Thanks!

**Comment by jackkamm** [comment]
For inferring the maximum likelihood demography, momi is able to handle missing data fine -- it doesn't throw out SNPs with missing data, and correctly accounts for the missing data when computing a SNP's likelihood.

However, when f4, IBS, and other model fitting statistics, and their Z-scores, momi doesn't yet handle missing data and discards SNPs with missing alleles. Hypergeometric project is implemented though, and you can use the second argument of `SfsModelFitStats` to specify a smaller SFS to project onto, as described [here](http://momi2.readthedocs.io/en/latest/tutorial.html#Limitations-of-SfsModelFitStats) and [here](http://momi2.readthedocs.io/en/latest/api.html#momi.SfsModelFitStats). 

There is no convenient function yet to import dadi data into momi. Despite many similarities, dadi and momi are targeted towards different regimes and as a consequence represent the SFS in very different ways -- in particular, momi is geared towards the many-population scenario, where the observed SFS is sparse (has many zeros), and thus uses a sparse data structure to represent the SFS.

Nevertheless it would be useful to convert dadi-style SFS into momi and I will add that as a feature request.

**Comment by isaacovercast** [comment]
Here's a gist with a function that[ converts dadi to momi style sfs:](https://gist.github.com/isaacovercast/01a3c76ccdcd197bee0bac6524ab0081). There's probably a smarter way to do it, but this is at least a place to start....

I can make a PR if you have an idea of the best way to incorporate the function into momi.

**Comment by jackkamm** [comment]
Thanks, please do make a PR, it would be great to have this functionality.

The /momi/data folder is where data reading functionality should be. I suggest making a new file, /momi/data/convert.py, and putting the function in there. Also, please add a small test, for example to test/test_data_structure.py.

**Comment by isaacovercast** [comment]
Ok, great. I am working on the convert.py code. Quick question about the Sfs.n_loci property. When you extract an Sfs from an allele_counts object you specify the number of "blocks", and this is used to determine the number of `n_loci` (indirectly, through the loc_idxs array). In the tutorial you use `100` for the number of blocks, but there's no indication of where this value came from. I understand this is used for resampling, and the optimal value is probably dependent on your data. For the purpose of constructing the Sfs from the dadi format, should I approach the blocks/n_loci question using a similar strategy as the `extract_sfs()` function, and allow the user to choose the number of blocks? Initially I assigned each snp to an independent locus, but now i'm not sure if this is right. What do you think?

**Comment by jackkamm** [comment]
I think best thing is to set `n_snps=1` (all SNPs on a single locus).

**Comment by isaacovercast** [comment]
I think this is fixed.
